### ISSUE TITLE
[BUG] r2 scores way too optimistic in .fit

When we run a fit on our tabular model, the r2 scores (or any other scores for that matter) that are published during the fit process are way too optimistic.  If we do a predict or  evaluate on the built model, we always get results which are worse than described in the .fit process.

### ISSUE THREAD

**Issue by alexkovaceski** [autogluon/autogluon#5113]
We suspect that the internal AG  train test split being performed for each bag/fold using default values when calling split and shuffling on that step is not actively being disabled. This is introducing data leakage and reducing the efficacy of the model.

https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html.

We request shuffle be explicitly disabled within the internal bagging/folding processes of Autogluon .fit.
